framework/src/problems/EigenProblem.C

```diff
@@ -540,8 +550,11 @@ EigenProblem::solve()
     _nl->solve();
     _nl->update();
 
-    // Scale eigen vector if users ask
-    postScaleEigenVector();
+    _first_solve = false;
```

Comment:
Doing this move is not necessary though. `_first_solve` is used only for initial free power iterations. Moving it to here makes things not so obvious.

Reply:
Motivation is that: if you not have "free_power", and then `_first_solve` will be always true. I like this change

Reply:
So you want `_first_solve` used somewhere else? Now I am thinking that we can wrap the free power iterations with `MooseDoOnce` then this flag is not visible.

Reply:
Like
```
      if (solverParams()._free_power_iterations)
      {
       MooseDoOnce(
        _console << " Free power iteration starts" << std::endl;
        doFreeNonlinearPowerIterations(solverParams()._free_power_iterations);
        );
      }
```